source/extensions/quic_listeners/quiche/envoy_quic_server_session.cc

```diff
@@ -103,17 +106,39 @@ void EnvoyQuicServerSession::OnCanWrite() {
 
 void EnvoyQuicServerSession::SetDefaultEncryptionLevel(quic::EncryptionLevel level) {
   quic::QuicServerSessionBase::SetDefaultEncryptionLevel(level);
-  if (level == quic::ENCRYPTION_FORWARD_SECURE) {
-    // This is only reached once, when handshake is done.
-    raiseConnectionEvent(Network::ConnectionEvent::Connected);
+  if (level != quic::ENCRYPTION_FORWARD_SECURE) {
+    return;
   }
+  // This is only reached once, when handshake is done.
+  raiseConnectionEvent(Network::ConnectionEvent::Connected);
+  maybeCreateNetworkFilters();
 }
 
 bool EnvoyQuicServerSession::hasDataToWrite() { return HasDataToWrite(); }
 
 void EnvoyQuicServerSession::OnOneRttKeysAvailable() {
   quic::QuicServerSessionBase::OnOneRttKeysAvailable();
   raiseConnectionEvent(Network::ConnectionEvent::Connected);
+  maybeCreateNetworkFilters();
+}
+
+void EnvoyQuicServerSession::maybeCreateNetworkFilters() {
+  const EnvoyQuicProofSourceDetails* proof_source_details =
+      dynamic_cast<const EnvoyCryptoServerStream*>(GetCryptoStream())->proofSourceDetails();
+  if (proof_source_details == nullptr) {
```

Comment:
Is this something worth fixing?  We do try to keep the unit tests accurate, even if we do something like mock out proofSourceDetails to return a mock pointer

Reply:
Makes sense, I added a MockFilterChain in TestProofSource. And ASSERT proof_source_details != nullptr here.